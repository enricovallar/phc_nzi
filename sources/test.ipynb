{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using MPI version 3.1, 1 processes\n",
      "Scheme configuration written to mpb_config.ctl\n",
      "(set! num-bands 8)\n",
      "(set! resolution 32)\n",
      "\n",
      "(set! k-points (list (vector3 0.0 0.0 0.0) (vector3 0.5 0.0 0.0) (vector3 0.5 0.5 0.0) (vector3 0.0 0.0 0.0)))\n",
      "(set! k-points (interpolate 4 k-points))\n",
      "(set! geometry (list\n",
      "    (make cylinder (center (vector3 0.0 0.0 0.0))(radius 0.2)(height 1e+20)(material (make dielectric (epsilon 12))))\n",
      "))\n",
      "(set! geometry-lattice (make lattice (size 1 1) (basis1  (vector3 1.0 0.0 0.0)) (basis2  (vector3 0.0 1.0 0.0)) ))\n",
      "(run-te)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import meep as mp\n",
    "from mpbConfigurator import *\n",
    "from photonicCrystal import *\n",
    "\n",
    " # Example usage\n",
    "lattice = Lattice(\"SX\", (1, 1))\n",
    "material = Material(epsilon=12)\n",
    "geometry = [Geometry(mp.Cylinder, {\"center\": mp.Vector3(0, 0), \"radius\": 0.2, \"height\": mp.inf, \"material\": material})]\n",
    "pc = PhotonicCrystal(geometry, lattice, material)\n",
    "\n",
    "\n",
    "# Define the MPB configuration\n",
    "config = MPBConfiguration(simulation_type=\"te\", resolution=32, num_bands=8, geometry_lattice=lattice, geometry=geometry)\n",
    "k_points = [mp.Vector3(0, 0, 0), mp.Vector3(0.5, 0, 0), mp.Vector3(0.5, 0.5, 0), mp.Vector3(0, 0, 0)]\n",
    "config.k_points = k_points\n",
    "\n",
    "\n",
    "# Generate the Scheme configuration\n",
    "scheme = config.generate_scheme_config(\"mpb_config.ctl\")\n",
    "print(scheme)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "init-params: initializing eigensolver data\n",
      "Computing 8 bands with 1.000000e-07 tolerance.\n",
      "Working in 2 dimensions.\n",
      "Grid size is 32 x 32 x 1.\n",
      "Solving for 8 bands at a time.\n",
      "Creating Maxwell data...\n",
      "Mesh size is 3.\n",
      "Lattice vectors:\n",
      "     (1, 0, 0)\n",
      "     (0, 1, 0)\n",
      "     (0, 0, 1)\n",
      "Cell volume = 1\n",
      "Reciprocal lattice vectors (/ 2 pi):\n",
      "     (1, -0, 0)\n",
      "     (-0, 1, -0)\n",
      "     (0, -0, 1)\n",
      "Geometric objects:\n",
      "     cylinder, center = (0,0,0)\n",
      "          radius 0.2, height 1e+20, axis (0, 0, 1)\n",
      "          epsilon = 12, mu = 1\n",
      "Geometric object tree has depth 1 and 1 object nodes (vs. 1 actual objects)\n",
      "Initializing epsilon function...\n",
      "Allocating fields...\n",
      "16 k-points:\n",
      "     (0,0,0)\n",
      "     (0.1,0,0)\n",
      "     (0.2,0,0)\n",
      "     (0.3,0,0)\n",
      "     (0.4,0,0)\n",
      "     (0.5,0,0)\n",
      "     (0.5,0.1,0)\n",
      "     (0.5,0.2,0)\n",
      "     (0.5,0.3,0)\n",
      "     (0.5,0.4,0)\n",
      "     (0.5,0.5,0)\n",
      "     (0.4,0.4,0)\n",
      "     (0.3,0.3,0)\n",
      "     (0.2,0.2,0)\n",
      "     (0.1,0.1,0)\n",
      "     (0,0,0)\n",
      "Solving for band polarization: te.\n",
      "Initializing fields to random numbers...\n",
      "elapsed time for initialization: 0 seconds.\n",
      "epsilon: 1-12, mean 2.38229, harm. mean 1.1448, 14.5508% > 1, 12.5663% \"fill\"\n",
      "Outputting mpb_config-epsilon...\n",
      "solve_kpoint (0,0,0):\n",
      "tefreqs:, k index, k1, k2, k3, kmag/2pi, te band 1, te band 2, te band 3, te band 4, te band 5, te band 6, te band 7, te band 8\n",
      "Solving for bands 2 to 8...\n",
      "Finished solving for bands 2 to 8 after 30 iterations.\n",
      "tefreqs:, 1, 0, 0, 0, 0, 0, 0.552709, 0.773227, 0.77323, 0.922997, 1.00017, 1.00017, 1.09282\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.1,0,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 16 iterations.\n",
      "tefreqs:, 2, 0.1, 0, 0, 0.1, 0.0897333, 0.551138, 0.7616, 0.774682, 0.907199, 1.00188, 1.00357, 1.11423\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.2,0,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 87 iterations.\n",
      "tefreqs:, 3, 0.2, 0, 0, 0.2, 0.178692, 0.544896, 0.728534, 0.778528, 0.880244, 1.00699, 1.01696, 1.10027\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.3,0,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 12 iterations.\n",
      "tefreqs:, 4, 0.3, 0, 0, 0.3, 0.265798, 0.528176, 0.685449, 0.783377, 0.860714, 1.01515, 1.03843, 1.09664\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.4,0,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 15 iterations.\n",
      "tefreqs:, 5, 0.4, 0, 0, 0.4, 0.348956, 0.490738, 0.652797, 0.787382, 0.849999, 1.02443, 1.06441, 1.09121\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.5,0,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 20 iterations.\n",
      "tefreqs:, 6, 0.5, 0, 0, 0.5, 0.412791, 0.441588, 0.641998, 0.788933, 0.846588, 1.02916, 1.08607, 1.08787\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.5,0.1,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 15 iterations.\n",
      "tefreqs:, 7, 0.5, 0.1, 0, 0.509902, 0.423741, 0.445781, 0.63823, 0.793147, 0.842943, 0.986523, 1.05018, 1.10383\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.5,0.2,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 12 iterations.\n",
      "tefreqs:, 8, 0.5, 0.2, 0, 0.538516, 0.454781, 0.457434, 0.627898, 0.805254, 0.826101, 0.927945, 1.00456, 1.10879\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.5,0.3,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 11 iterations.\n",
      "tefreqs:, 9, 0.5, 0.3, 0, 0.583095, 0.473767, 0.501279, 0.61358, 0.781467, 0.824181, 0.894949, 0.959051, 1.1107\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.5,0.4,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 10 iterations.\n",
      "tefreqs:, 10, 0.5, 0.4, 0, 0.640312, 0.489649, 0.556045, 0.59957, 0.719611, 0.849247, 0.884337, 0.917168, 1.11162\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.5,0.5,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 9 iterations.\n",
      "tefreqs:, 11, 0.5, 0.5, 0, 0.707107, 0.496836, 0.593141, 0.593154, 0.679169, 0.878684, 0.881977, 0.881982, 1.11192\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.4,0.4,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 10 iterations.\n",
      "tefreqs:, 12, 0.4, 0.4, 0, 0.565685, 0.470987, 0.546003, 0.606732, 0.743602, 0.839223, 0.882254, 0.93697, 1.11121\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.3,0.3,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 12 iterations.\n",
      "tefreqs:, 13, 0.3, 0.3, 0, 0.424264, 0.372435, 0.536186, 0.644572, 0.809689, 0.825804, 0.883642, 1.0102, 1.10725\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.2,0.2,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 15 iterations.\n",
      "tefreqs:, 14, 0.2, 0.2, 0, 0.282843, 0.252088, 0.542887, 0.698081, 0.789371, 0.888483, 0.897858, 1.07884, 1.09039\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0.1,0.1,0):\n",
      "Solving for bands 1 to 8...\n",
      "Finished solving for bands 1 to 8 after 18 iterations.\n",
      "tefreqs:, 15, 0.1, 0.1, 0, 0.141421, 0.126841, 0.549908, 0.750378, 0.777261, 0.903059, 0.961984, 1.04191, 1.10004\n",
      "elapsed time for k point: 0 seconds.\n",
      "solve_kpoint (0,0,0):\n",
      "Solving for bands 2 to 8...\n",
      "Finished solving for bands 2 to 8 after 84 iterations.\n",
      "tefreqs:, 16, 0, 0, 0, 0, 0, 0.552709, 0.773227, 0.77323, 0.922997, 1.00017, 1.00017, 1.09282\n",
      "elapsed time for k point: 0 seconds.\n",
      "Band 1 range: 0.0 at #(0.0 0.0 0.0) to 0.4968358657398469 at #(0.5 0.5 0.0)\n",
      "Band 2 range: 0.44158847583430555 at #(0.5 0.0 0.0) to 0.5931405182468622 at #(0.5 0.5 0.0)\n",
      "Band 3 range: 0.5931535893552098 at #(0.5 0.5 0.0) to 0.773226567857314 at #(0.0 0.0 0.0)\n",
      "Band 4 range: 0.6791690135255452 at #(0.5 0.5 0.0) to 0.8096891662050567 at #(0.30000000000000004 0.30000000000000004 0.0)\n",
      "Band 5 range: 0.8241814492203634 at #(0.5 0.30000000000000004 0.0) to 0.9229965309504361 at #(0.0 0.0 0.0)\n",
      "Band 6 range: 0.8819770936730511 at #(0.5 0.5 0.0) to 1.0291597307275677 at #(0.5 0.0 0.0)\n",
      "Band 7 range: 0.8819818143162845 at #(0.5 0.5 0.0) to 1.0860733987653632 at #(0.5 0.0 0.0)\n",
      "Band 8 range: 1.087869087126774 at #(0.5 0.0 0.0) to 1.1142329836202252 at #(0.10000000000000003 0.0 0.0)\n",
      "Gap from band 2 (0.5931405182468622) to band 3 (0.5931535893552098), 0.0022036876460603374%\n",
      "Gap from band 4 (0.8096891662050567) to band 5 (0.8241814492203634), 1.7739817190522467%\n",
      "Gap from band 7 (1.0860733987653632) to band 8 (1.087869087126774), 0.16520109184708645%\n",
      "total elapsed time for run: 0 seconds.\n",
      "done.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import subprocess\n",
    "\n",
    "# Run the simulation\n",
    "filename = \"mpb_config.ctl\" \n",
    "command = [\"mpb\", filename]\n",
    "result = subprocess.run(command,  capture_output=True, text=True)\n",
    "print(result.stdout)\n",
    "print(result.stderr)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "nzi-mp",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
