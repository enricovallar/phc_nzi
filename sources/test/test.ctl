(set! num-bands 6)
(set! resolution 32)
(set! k-points (list (vector3 0.0 0.0 0.0) (vector3 0.3333333333333333 0.0 0.0) (vector3 0.5 0.5 0.0) (vector3 0.0 0.0 0.0)))
(set! k-points (interpolate 10 k-points))
(define-param r 0.25)
(define-param r2 0.15)
(set! geometry-lattice (make lattice (size 1 1 no-size)  (basis1  (vector3 1.0 0.0 0.0)) (basis2  (vector3 0.5000000000000001 0.8660254037844388 0.0)) ))
(set! geometry (list 
 (make block (size (vector3 1e+20 1e+20 1e+20))
  (material (make dielectric (epsilon 9.61)))
  (center (vector3 0.0 0.0 0.0)))
 (make cylinder (radius r)
  (height 1e+20)
  (material (make dielectric (epsilon 1)))
  (center (vector3 0.3333333333333333 0.3333333333333333 0.0)))
 (make cylinder (radius r2)
  (height 1e+20)
  (material (make dielectric (epsilon 1)))
  (center (vector3 0.6666666666666666 0.6666666666666666 0.0)))
))
(run-te)
(run-tm)
