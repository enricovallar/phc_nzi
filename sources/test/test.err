Modules Release 4.8.0 (2021-07-14)
DCC-SW: Added modules (2023-aug/XeonGold6342)
DCC-SW: Removed standard modules: /apps/dcc/etc/ModulesAlma92/modulefiles
DCC-SW: Using gnu/12.3.0 compiler (requested)
Loading mpb/1.11.1
  Loading requirement: build-tools/1.0 gcc/12.3.0 zlib/2.0.7 numactl/2.0.16
    libxml2/2.10.3 hwloc/2.9.0 ucx/1.14.0 openmpi/4.1.6 libctl/4.5.1 szip/2.1.1
    hdf5/1.14.0 fftw-mpi/3.3.10 lapack/3.12.0 blis/1.0
