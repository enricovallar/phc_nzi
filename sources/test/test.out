init-params: initializing eigensolver data
Computing 8 bands with 1.000000e-07 tolerance.
Working in 2 dimensions.
Grid size is 32 x 32 x 1.
Solving for 8 bands at a time.
Creating Maxwell data...
Mesh size is 3.
Lattice vectors:
     (1, 0, 0)
     (0.5, 0.866025, 0)
     (0, 0, 1)
Cell volume = 0.866025
Reciprocal lattice vectors (/ 2 pi):
     (1, -0.57735, 0)
     (-0, 1.1547, -0)
     (0, -0, 1)
Geometric objects:
     cylinder, center = (0,0,0)
          radius 0.2, height 1e+20, axis (0, 0, 1)
          epsilon = 12, mu = 1
Geometric object tree has depth 1 and 1 object nodes (vs. 1 actual objects)
Initializing epsilon function...
Allocating fields...
16 k-points:
     (0,0,0)
     (0.1,0,0)
     (0.2,0,0)
     (0.3,0,0)
     (0.4,0,0)
     (0.5,0,0)
     (0.5,0.1,0)
     (0.5,0.2,0)
     (0.5,0.3,0)
     (0.5,0.4,0)
     (0.5,0.5,0)
     (0.4,0.4,0)
     (0.3,0.3,0)
     (0.2,0.2,0)
     (0.1,0.1,0)
     (0,0,0)
Solving for band polarization: te.
Initializing fields to random numbers...
elapsed time for initialization: 0 seconds.
epsilon: 1-12, mean 2.59614, harm. mean 1.17149, 17.2852% > 1, 14.5104% "fill"
Outputting test-epsilon...
solve_kpoint (0,0,0):
tefreqs:, k index, k1, k2, k3, kmag/2pi, te band 1, te band 2, te band 3, te band 4, te band 5, te band 6, te band 7, te band 8
Solving for bands 2 to 8...
Finished solving for bands 2 to 8 after 32 iterations.
tefreqs:, 1, 0, 0, 0, 0, 0, 0.561229, 0.793929, 0.794659, 1.03341, 1.03342, 1.11597, 1.11681
elapsed time for k point: 0 seconds.
solve_kpoint (0.1,0,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 14 iterations.
tefreqs:, 2, 0.1, 0, 0, 0.11547, 0.101865, 0.559805, 0.79146, 0.793258, 0.992126, 1.02845, 1.09799, 1.1151
elapsed time for k point: 0 seconds.
solve_kpoint (0.2,0,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 14 iterations.
tefreqs:, 3, 0.2, 0, 0, 0.23094, 0.202662, 0.555133, 0.784265, 0.784932, 0.925328, 1.01804, 1.06979, 1.11298
elapsed time for k point: 0 seconds.
solve_kpoint (0.3,0,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 11 iterations.
tefreqs:, 4, 0.3, 0, 0, 0.34641, 0.300756, 0.545003, 0.752836, 0.776604, 0.881528, 1.00851, 1.04494, 1.1117
elapsed time for k point: 0 seconds.
solve_kpoint (0.4,0,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 10 iterations.
tefreqs:, 5, 0.4, 0, 0, 0.46188, 0.392756, 0.521577, 0.705571, 0.77011, 0.86865, 1.00251, 1.02766, 1.11107
elapsed time for k point: 0 seconds.
solve_kpoint (0.5,0,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 9 iterations.
tefreqs:, 6, 0.5, 0, 0, 0.57735, 0.468428, 0.473072, 0.683225, 0.767662, 0.866987, 1.00052, 1.02137, 1.11088
elapsed time for k point: 0 seconds.
solve_kpoint (0.5,0.1,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 16 iterations.
tefreqs:, 7, 0.5, 0.1, 0, 0.52915, 0.44163, 0.504537, 0.674011, 0.77871, 0.867032, 0.952535, 1.06781, 1.1107
elapsed time for k point: 0 seconds.
solve_kpoint (0.5,0.2,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 14 iterations.
tefreqs:, 8, 0.5, 0.2, 0, 0.503322, 0.425069, 0.530136, 0.659329, 0.798468, 0.867241, 0.910994, 1.09951, 1.1096
elapsed time for k point: 0 seconds.
solve_kpoint (0.5,0.3,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 14 iterations.
tefreqs:, 9, 0.5, 0.3, 0, 0.503322, 0.425074, 0.530125, 0.659354, 0.798735, 0.867265, 0.910701, 1.09939, 1.10951
elapsed time for k point: 0 seconds.
solve_kpoint (0.5,0.4,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 16 iterations.
tefreqs:, 10, 0.5, 0.4, 0, 0.52915, 0.44164, 0.504525, 0.674042, 0.77924, 0.867065, 0.951975, 1.0674, 1.11058
elapsed time for k point: 0 seconds.
solve_kpoint (0.5,0.5,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 15 iterations.
tefreqs:, 11, 0.5, 0.5, 0, 0.57735, 0.468489, 0.473019, 0.683232, 0.768263, 0.867023, 0.999706, 1.0211, 1.11074
elapsed time for k point: 0 seconds.
solve_kpoint (0.4,0.4,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 8 iterations.
tefreqs:, 12, 0.4, 0.4, 0, 0.46188, 0.392779, 0.521573, 0.705494, 0.770706, 0.868733, 1.00173, 1.02743, 1.11093
elapsed time for k point: 0 seconds.
solve_kpoint (0.3,0.3,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 10 iterations.
tefreqs:, 13, 0.3, 0.3, 0, 0.34641, 0.300786, 0.545005, 0.752525, 0.777191, 0.881767, 1.00784, 1.04478, 1.11152
elapsed time for k point: 0 seconds.
solve_kpoint (0.2,0.2,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 12 iterations.
tefreqs:, 14, 0.2, 0.2, 0, 0.23094, 0.202688, 0.555136, 0.783898, 0.785357, 0.925669, 1.01758, 1.06978, 1.1127
elapsed time for k point: 0 seconds.
solve_kpoint (0.1,0.1,0):
Solving for bands 1 to 8...
Finished solving for bands 1 to 8 after 14 iterations.
tefreqs:, 15, 0.1, 0.1, 0, 0.11547, 0.10188, 0.559807, 0.792072, 0.792653, 0.992333, 1.02827, 1.09825, 1.11463
elapsed time for k point: 1 seconds.
solve_kpoint (0,0,0):
Solving for bands 2 to 8...
Finished solving for bands 2 to 8 after 16 iterations.
tefreqs:, 16, 0, 0, 0, 0, 0, 0.561229, 0.793929, 0.794659, 1.03341, 1.03342, 1.11597, 1.11681
elapsed time for k point: 0 seconds.
Band 1 range: 0.0 at #(0.0 0.0 0.0) to 0.46848875197064394 at #(0.5 0.5 0.0)
Band 2 range: 0.47301908783017593 at #(0.5 0.5 0.0) to 0.5612286894375408 at #(0.0 0.0 0.0)
Band 3 range: 0.6593292810495819 at #(0.5 0.20000000000000004 0.0) to 0.7939289546108732 at #(0.0 0.0 0.0)
Band 4 range: 0.7676615135183539 at #(0.5 0.0 0.0) to 0.7987352819166326 at #(0.5 0.30000000000000004 0.0)
Band 5 range: 0.8669873636678131 at #(0.5 0.0 0.0) to 1.0334130564345925 at #(0.0 0.0 0.0)
Band 6 range: 0.9107009844601726 at #(0.5 0.30000000000000004 0.0) to 1.033423128163494 at #(0.0 0.0 0.0)
Band 7 range: 1.0211008414688996 at #(0.5 0.5 0.0) to 1.1159670894103342 at #(0.0 0.0 0.0)
Band 8 range: 1.1095085572205043 at #(0.5 0.30000000000000004 0.0) to 1.116811157238839 at #(0.0 0.0 0.0)
Gap from band 1 (0.46848875197064394) to band 2 (0.47301908783017593), 0.962357543510293%
Gap from band 2 (0.5612286894375408) to band 3 (0.6593292810495819), 16.074712383040573%
Gap from band 4 (0.7987352819166326) to band 5 (0.8669873636678131), 8.194891500347376%
total elapsed time for run: 1 seconds.
done.
